frontend/test/resolution/testIteratorsToArrays.cpp

```diff
@@ -0,0 +1,270 @@
+  testArrayMaterialize(context,
+                       "",
+                       "for i in {1..10} do i",
+                       1, "one", "ChapelArray.chpl__initCopy#4");
+  testArrayMaterialize(context,
+                       "",
+                       "for i in {1..10, 1..10} do i",
+                       2, "one", "ChapelArray.chpl__initCopy#4");
+  testArrayMaterialize(context,
+                       "",
+                       "for i in {1..10 by 2, 1..10} do i",
+                       2, "positive", "ChapelArray.chpl__initCopy#4");
+}
+
+static void testPromotionRange(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(1..10)",
+                       1, "one", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionStridedRange(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(1..10 by 2)",
+                       1, "positive", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionReverseRange(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(10..1 by -1)",
+                       1, "negOne", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionReverseStridedRange(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(10..1 by -2)",
+                       1, "negative", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionOverPromotion(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(foo(1..10))",
+                       1, "one", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionOverStridedPromotion(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo(foo(1..10 by 2))",
+                       1, "positive", "ChapelArray.chpl__initCopy#5");
+}
+
+static void testPromotionOverDomain(Context* context) {
+  testArrayMaterialize(context,
+                       "proc foo(x) do return x : real;",
+                       "foo({1..10})",
+                       1, "one", "ChapelArray.chpl__initCopy#4");
+  testArrayMaterialize(context,
+                       "proc foo((x, y)) do return (x : real, y : real);",
+                       "foo({1..10, 1..10})",
+                       2, "one", "ChapelArray.chpl__initCopy#4");
+  testArrayMaterialize(context,
+                       "proc foo((x, y)) do return (x : real, y : real);",
+                       "foo({1..10 by 2, 1..10})",
+                       2, "positive", "ChapelArray.chpl__initCopy#4");
+}
+
+int main() {
+  auto context = buildStdContext();
```

Comment:
I tried it out and found no measurable performance difference between reusing a context, or creating a new one inside `testArrayMaterialize` each time. So you might want to remove it to simplify these test functions a tiny bit, though I don't think it matters much either way.

Also, at least my `testDomains` isn't the longest running dyno test :p